src/solve.ml

```diff
@@ -140,11 +156,13 @@ let string_of_smt_query spec m1 m2 get_vals smt_exp = (* The query used in valid
 let smt_bowtie = EVar(Var("bowtie"))
 let smt_oper = EVar(Var("oper"))
 
-let commute_of_smt smt = EBop(Imp, ELop(And, [smt_oper; smt]), smt_bowtie)
-let commute spec h = smt_of_conj (add_conjunct spec.precond h) |> commute_of_smt
+(* let commute_of_smt spec smt = EBop(Imp, ELop(And, [smt_oper; smt]), ELop(And, [spec.postcond; smt_bowtie])) *)
+let commute_of_smt spec smt = EBop(Imp, ELop(And, [smt_oper; smt]), smt_bowtie)
```

Comment:
spec isn't used in the body? Why add it as a parameter?

Reply:
Was discussed in meeting. Remove the parameter here and where the function is called.